tests/test_swift.py

```diff
@@ -87,35 +73,139 @@ def setUp(self):
         for channel in self.channels:
             self.assertIn(channel, self.swift._subscribers)
 
+    def test_app_names(self):
+        appNames = self.swift.appNames()
+        for name in APP_INFOS:
+            self.assertIn(name, appNames)
+
     def test_destroy_app(self):
         for name in APP_INFOS:
             self.swift.destroyApp(name)
             self.assertNotIn(name, self.swift._apps)
 
+    def test_update_frames(self):
+        self.swift._apps["app1"].frames.return_value = (QWidget(),)
+        self.swift.updateFrames("app1")
+
+    def test_channel_names(self):
+        channelNames = self.swift.channelNames()
+        for channelName in self.channels:
+            self.assertIn(channelName, channelNames)
+
+    def test_subscriber_names(self):
+        for channel in self.channels:
+            subscriberNames = self.swift.subscriberNames(channel)
+            for name, info in APP_INFOS.items():
+                if channel in info.channel:
+                    self.assertIn(name, subscriberNames)
+
+    def test_unsubcribe(self):
+        self.assertEqual(self.swift.unsubscribe("app1", "ch1"), True)
+        self.assertNotIn("app1", self.swift._subscribers["ch1"])
+        self.assertEqual(self.swift.unsubscribe("app2", "ch1"), False)
+
+    def test_swiftcall(self):
+        QMessageBox.warning = MagicMock(return_value=QMessageBox.Ok)
+        self.swift._swiftcall("app1", json.dumps({"call": "appNames", "args": {}}))
+        self.swift._swiftcall("app1", json.dumps({
+            "call": "createApp",
+            "args": {"name": "app3", "info": {"module": "module2", "cls": "cls2"}}
+        }))
+        self.swift.callWithSerializable = MagicMock()
+        self.swift.getSerializable = MagicMock(return_value=APP_INFOS["app1"])
+        self.swift._swiftcall("app1", json.dumps({"call": "getSerializable", "args": {}}))
+        self.swift._swiftcall("app1", json.dumps({"call": "_broadcast", "args": {}}))
+        QMessageBox.warning.return_value = QMessageBox.Cancel
+        self.swift._swiftcall("app1", json.dumps({"call": "appNames", "args": {}}))
+
     def test_broadcast(self):
         for channelName in self.channels:
             self.swift._broadcast(channelName, "test_msg")
         for name, app_ in self.swift._apps.items():
             self.assertEqual(len(APP_INFOS[name].channel), app_.received.emit.call_count)
 
 
+class BaseAppTest(unittest.TestCase):
+    """Unit test for BaseApp class."""
+
+    def setUp(self):
+        self.app = swift.BaseApp("name")
+
+    def test_init(self):
+        self.assertEqual(self.app.name, "name")
+        self.assertIsInstance(self.app.swiftcall, swift.SwiftcallProxy)
+        self.assertEqual(type(self.app.broadcastRequested), pyqtBoundSignal)
+        self.assertEqual(type(self.app.received), pyqtBoundSignal)
+        self.assertEqual(type(self.app.swiftcallRequested), pyqtBoundSignal)
+        self.assertEqual(type(self.app.swiftcallReturned), pyqtBoundSignal)
```

Comment:
Is this necessary..? Is it for the coverage?

Reply:
Oh, it isn't for the coverage.
I just checked if the types of the fields are correct.

If you think it isn't necessary, I will remove it.

Reply:
I think it is not necessary, because it is not our responsibility to check their type, but `PyQt5`'s.
We just used the `pyqtSignal()` so it should be the type that `PyQt5` package specifies.

Reply:
Okay! Then, how about `test_init()` of `SwiftTest`?

Reply:
Oh yeah those things as well 😄
Let's not instance-check them!